geospaas_harvesting/verify.py

```diff
@@ -0,0 +1,35 @@
+""" Verification module. """
+import os
+import sys
+from datetime import datetime
+
+import django
+import requests
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_harvesting.settings')
+django.setup()
+
+from geospaas.catalog.models import DatasetURI
+
+def main(filename):
+    """ Verifies the datasets based on their dataseturi. If the download link does not provide a
+    download availability and returns a text or html response, then the dataset uri is removed. If
+    there is no other "dataseturi" remains for the dataset, then the dataset is also removed in
+    order to be harvested again in the future with a correct and healthy "dataseturi".
+    Since the number of datasets in the database might be enormous, the datasets are retrieved
+    into with a variable named retrieved_dataset_uris with an specific length number
+    for memory management. """
+    if filename=='':
+        filename=f"unverified_dataset_at_{datetime.now().strftime('%Y-%m-%d___%H_%M_%S')}"
+    with open(filename+".txt", 'w') as f:
+        for dsuri in DatasetURI.objects.iterator(chunk_size=1000).__iter__():
```

Comment:
- Please read about python iterators. They are not supposed to be used like that.
- As explained in the documentation:
  "The default value of **chunk_size**, 2000, comes from a calculation on the psycopg mailing list".
  Let's use the default value unless we have a good reason not to.

```suggestion
        for dsuri in DatasetURI.objects.iterator():
```

Reply:
Done with github online version. I will read about it more.